TreeMaker/python/makeJetVars.py

```diff
@@ -172,6 +174,28 @@ def makeJetVars(self, process, JetTag, suff, storeProperties, SkipTag=cms.VInput
             JetTag       = GoodJetsTag
         )
 
+        if storePileupIds:
+            from TreeMaker.Utils.pileupjetid_cfi import pileupJetId, pileupjetalgos
+            pileupJetIdUpdated = pileupJetId.clone(
+                jets = GoodJetsTag,
+                inputIsCorrected=True,
+                applyJec=False,
+                vertexes = cms.InputTag("offlineSlimmedPrimaryVertices"),
+                algos=pileupjetalgos["2016UL"]
+            )
+            (TMeras.TMUL2016APV).toModify(pileupJetIdUpdated,algos = pileupjetalgos["2016UL_APV"])
+            (TMeras.TMUL2017).toModify(pileupJetIdUpdated,algos = pileupjetalgos["2017UL"])
+            (TMeras.TMUL2018).toModify(pileupJetIdUpdated,algos = pileupjetalgos["2018UL"])
+
+            setattr(process,"pileupJetId"+suff,pileupJetIdUpdated)
+
+            JetProperties.properties.extend(["pileupJetId"])
+            JetProperties.pileupJetId = cms.vstring('pileupJetId' + suff + ':fullDiscriminant')

```

Comment:
have you tested that this actually works? the jet collection is never updated to include this userfloat...

Reply:
I have run some of the tests and it does produce a filled branch in the NTuple. One point I wanted to ask is about the use of HUGE_VAL for producing an "invalid value," since PileupID is only valid for jets with pt<50, and produces a meaningless value for pt>50. Perhaps better would be std::optional<float> (if TreeMaker supports it). Or I could also just provide no check the user would need to enforce this in their analysis framework. Please let me know what you think.

Reply:
I suspect it is just picking up the default userfloat from the miniAOD, since your new userfloat is not included in any jet collections. If what is in the miniAOD uses the correct training, then `pileupJetIdUpdated` is not needed here and can just be removed. If what is in the miniAOD does not use the correct training, then you need to update the jet collection with the new userfloat.

Reply:
Hi Kevin. Sorry, I am not very familiar with this, but the twiki seems to suggest that the discriminant can be accessed through the userfloat. https://twiki.cern.ch/twiki/bin/view/CMS/PileupJetIDUL . I am not sure how to add it manually.

Reply:
Ok, I believe I have resolved this issue. As a check I have the histograms for unittest 2 before and after commit [ad0d760](https://github.com/TreeMaker/TreeMaker/pull/652/commits/ad0d7607749273fff80f931e6fa163d37bd145f5).

<img width="679" alt="Pileup_Pre_PR" src="https://user-images.githubusercontent.com/50681385/195105039-3ed3b962-94b9-4ac7-9d9a-e8d265d08a75.png">
ved this issue. 
<img width="601" alt="PileUpID_Post_PR" src="https://user-images.githubusercontent.com/50681385/195105054-5786244a-c7a0-44ab-8ed6-824068c98d28.png">